url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/EventSetDescriptor.java

```diff
@@ -0,0 +1,432 @@
+     */
+    private static String extractShortClassName(String fullClassName) {
+        int k = fullClassName.lastIndexOf('$');
+        k = (k == -1 ? fullClassName.lastIndexOf('.') : k);
+        return fullClassName.substring(k + 1);
+    }
+
+    private static String prepareEventTypeName(String eventSetName) {
+        StringBuilder sb = new StringBuilder();
+
+        if (eventSetName != null && eventSetName.length() > 0) {
+            sb.append(Character.toUpperCase(eventSetName.charAt(0)));
+
+            if (eventSetName.length() > 1) {
+                sb.append(eventSetName.substring(1));
+            }
+        }
+
+        sb.append("Event"); //$NON-NLS-1$
+        return sb.toString();
+    }
+
+    public Method[] getListenerMethods() {
+        if (listenerMethods != null) {
+            return listenerMethods;
+        }
+
+        if (listenerMethodDescriptors != null) {
+            listenerMethods = new Method[listenerMethodDescriptors.size()];
+            int index = 0;
+            for (MethodDescriptor md : listenerMethodDescriptors) {
+                listenerMethods[index++] = md.getMethod();
+            }
+            return listenerMethods;
+        }
+        return null;
+    }
+
+    public MethodDescriptor[] getListenerMethodDescriptors() {
+        return listenerMethodDescriptors == null ? null
+                : listenerMethodDescriptors.toArray(new MethodDescriptor[0]);
+    }
+
+    public Method getRemoveListenerMethod() {
+        return removeListenerMethod;
+    }
+
+    public Method getGetListenerMethod() {
+        return getListenerMethod;
+    }
+
+    public Method getAddListenerMethod() {
+        return addListenerMethod;
+    }
+
+    public Class<?> getListenerType() {
+        return listenerType;
+    }
+
+    public void setUnicast(boolean unicast) {
+        this.unicast = unicast;
+    }
+
+    public void setInDefaultEventSet(boolean inDefaultEventSet) {
+        this.inDefaultEventSet = inDefaultEventSet;
+    }
+
+    public boolean isUnicast() {
+        return unicast;
+    }
+
+    public boolean isInDefaultEventSet() {
+        return inDefaultEventSet;
+    }
+
+    /**
+     * Searches for {add|remove}Listener methods in the event source. Parameter
+     * check is also performed.
+     * 
+     * @param sourceClass
+     *            event source class
+     * @param methodName
+     *            method name to search
+     * @return found method
+     * @throws IntrospectionException
+     *             if no valid method found
+     */
+    private Method findAddRemoveListenerMethod(Class<?> sourceClass,
+            String methodName) throws IntrospectionException {
+        try {
+            return sourceClass.getMethod(methodName, listenerType);
+        } catch (NoSuchMethodException e) {
+            return findAddRemoveListnerMethodWithLessCheck(sourceClass,
+                    methodName);
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)